plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/ConstraintExtractor.java

```diff
@@ -316,6 +336,72 @@ private static Optional<Domain> unwrapDateTruncInComparison(String unit, Functio
         return Optional.empty();
     }
 
+    private static Optional<TupleDomain<IcebergColumnHandle>> unwrapYearInComparison(
+            // upon invocation, we don't know if this really is a comparison
+            FunctionName functionName,
+            ConnectorExpression yearSource,
+            Constant constant,
+            Map<String, ColumnHandle> assignments)
+    {
+        if (!(yearSource instanceof Variable sourceVariable)) {
+            // Engine unwraps year in comparisons in UnwrapYearInComparison. Within a connector we can do more than
+            // engine only for source columns. We cannot draw many conclusions for intermediate expressions without
+            // knowing them well.
+            return Optional.empty();
+        }
+
+        if (constant.getValue() == null) {
+            // Comparisons with NULL should be simplified by the engine
+            return Optional.empty();
+        }
+
+        IcebergColumnHandle column = resolve(sourceVariable, assignments);
+        if (column.getType() instanceof TimestampWithTimeZoneType type) {
+            // Iceberg supports only timestamp(6) with time zone
+            checkArgument(type.getPrecision() == 6, "Unexpected type: %s", column.getType());
+
+            return unwrapYearInComparison(functionName, type, constant)
+                    .map(domain -> TupleDomain.withColumnDomains(ImmutableMap.of(column, domain)));
+        }
+
+        return Optional.empty();
+    }
+
+    private static Optional<Domain> unwrapYearInComparison(FunctionName functionName, Type argumentType, Constant constant)
+    {
+        checkArgument(constant.getValue() != null && argumentType.equals(TIMESTAMP_TZ_MICROS), "Unexpected constant: %s", constant);
+
+        // Normalized to UTC because for comparisons the zone is irrelevant
+        ZonedDateTime periodStart = ZonedDateTime.of(LocalDate.ofYearDay(toIntExact((Long) constant.getValue()), 1), LocalTime.MIDNIGHT, UTC);
+        ZonedDateTime periodEnd = periodStart.with(lastDayOfYear()).with(LocalTime.MAX);
+
+        LongTimestampWithTimeZone start = LongTimestampWithTimeZone.fromEpochSecondsAndFraction(periodStart.toEpochSecond(), 0, UTC_KEY);
+        LongTimestampWithTimeZone end = LongTimestampWithTimeZone.fromEpochSecondsAndFraction(periodEnd.toEpochSecond(), MIDNIGHT_FRACTION_IN_PICOS, UTC_KEY);
```

Comment:
right-open ranges are easier to construct and work with, see `unwrapTimestampTzToDateCast`

in fact, there is high overlap between that method and the new one, opportunity for re-use